File: windows/src/GoalMaker.Core/Settings/Appearance.cs
namespace GoalMaker.Core.Settings;

/// <summary>
/// How the app looks on this PC (docs/design/spec.md, "Settings"). Not synced. <see cref="ThemeId"/>
/// names a theme in contracts/design/themes.json; unknown ids fall back to the default theme.
/// </summary>
public sealed record Appearance(string? ThemeId, ThemeMode Mode, bool PureBlack, ReduceMotion ReduceMotion, bool CompletionSound)
{
    /// <summary>A fresh install: the default theme, following Windows' light or dark.</summary>
    public static Appearance Default { get; } = new(null, ThemeMode.System, PureBlack: false, ReduceMotion.System, CompletionSound: false);
}

File: android/app/src/main/kotlin/com/goalmaker/app/domain/settings/Appearance.kt
package com.goalmaker.app.domain.settings

/**
 * How the app looks on this device (docs/design/spec.md, "Settings"). Not synced: each device keeps
 * its own. [themeId] names a theme in contracts/design/themes.json; unknown ids fall back to the
 * default theme.
 */
data class Appearance(
    val themeId: String?,
    val mode: ThemeMode,
    val pureBlack: Boolean,
    val reduceMotion: ReduceMotion,
    val completionSound: Boolean,
) {
    companion object {
        /** A fresh install: the default theme, following the system's light or dark. */
        val DEFAULT = Appearance(
            themeId = null,
            mode = ThemeMode.SYSTEM,
            pureBlack = false,
            reduceMotion = ReduceMotion.SYSTEM,
            completionSound = false,
        )
    }
}
